sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.c

```diff
@@ -0,0 +1,410 @@
+            assert(node->mode & _DRL);
+            if ((last == NULL)) {
+                return def_router;
+            }
+            else if (!found && (last == def_router)) {
+                found = true;
+            }
+            else if (found) {
+                return def_router;
+            }
+        }
+    }
+    return NULL;
+}
+
+_nib_dr_t *_nib_drl_get(const ipv6_addr_t *router_addr, unsigned iface)
+{
+    for (unsigned i = 0; i < GNRC_IPV6_NIB_DEFAULT_ROUTER_NUMOF; i++) {
+        _nib_dr_t *def_router = &_def_routers[i];
+        _nib_t *node = def_router->next_hop;
+
+        if ((node != NULL) &&
+            (ipv6_addr_equal(router_addr, &node->ipv6)) &&
+            (_nib_get_if(node) == iface)) {
+            /* It is linked to the default router list so it *should* be set */
+            assert(node->mode & _DRL);
+            return def_router;
+        }
+    }
+    return NULL;
+}
+
+_nib_dr_t *_nib_drl_get_dr(void)
+{
+    if ((_prime_def_router == NULL) ||
+        (_node_unreachable(_prime_def_router->next_hop))) {
+        _prime_def_router = NULL;   /* in case the prime router was unreachable */
+        _nib_dr_t *last = _last_def_router;
+        _nib_dr_t *res = _nib_drl_iter(last);
+
+        if (res == NULL) {
+            res = _nib_drl_iter(res);   /* try to wrap around */
+            if (res == NULL) {          /* we really don't have a default router */
+                return NULL;
+            }
+        }
+        /* search a reachable router */
+        while (_node_unreachable(res->next_hop)) {
+            last = res;
+            res = _nib_drl_iter(res);       /* go further */
+            if (res == NULL) {
+                res = _nib_drl_iter(res);   /* wrap around */
+                if (_last_def_router == NULL) {
+                    /* we iterated over the complete list, since we started at
+                     * the first element */
+                    break;
+                }
+            }
+            if (last == _last_def_router) {
+                break;
+            }
+        }
+        if (_node_unreachable(res->next_hop)) {
+            _last_def_router = res;
+        }
+        else {
+            _last_def_router = NULL;
+            _prime_def_router = res;
+        }
+        return res;
+    }
+    return _prime_def_router;
+}
+
+_nib_dst_t *_nib_dc_add(const ipv6_addr_t *next_hop, unsigned iface,
+                        const ipv6_addr_t *dst)
+{
+    _nib_dst_t *nib_dst = _nib_dst_alloc(next_hop, iface, dst, IPV6_ADDR_BIT_LEN);
+    if (nib_dst != NULL) {
+        nib_dst->next_hop->mode |= (_DC);
+    }
+    return nib_dst;
+}
+
+_nib_iface_t *_nib_iface_get(unsigned iface)
+{
+    _nib_iface_t *ni = NULL;
+
+    assert((iface > KERNEL_PID_UNDEF) && (iface <= KERNEL_PID_LAST));
+    for (unsigned i = 0; i < GNRC_NETIF_NUMOF; i++) {
+        _nib_iface_t *tmp = &_nis[i];
+        if (((unsigned)tmp->pid) == iface) {
```

Comment:
The range for interfaces was selected from the range of `kernel_pid_t`, defined by `KERNEL_PID_LAST`.